openghg/util/_file.py

```diff
@@ -247,27 +248,31 @@ def get_logfile_path() -> Path:
         return Path("/tmp/openghg.log")
 
 
-def check_function_open_nc(filepath: multiPathType) -> Tuple[Callable, multiPathType]:
+def open_and_align_dataset(filepath: multiPathType, domain: str) -> Tuple[xr.Dataset, multiPathType]:
```

Comment:
Rather than replacing this function (which is used more widely than just for footprints), it would be good to create this as a separate function as an alternative option for footprints to use.

Updating this to restore the `check_function_open_nc` may also go some way towards getting past the `mypy` issues which are coming up as well.

Reply:
To keep the same behaviour, you could use e.g. `xr_open_fn = partial(xr.open_mfdataset, preprocess=lambda x: align_lat_lon(x, domain))`.

On the other hand, is there a reason for returning the function to call and the filepath(s) separately? It might make sense to just do the final `xr_open_fn(filepath)` in this function.

Reply:
Are you sure that `check_function_open_nc` is used elsewhere @rt17603 ? I cannot find it anywhere else when I search for it, that's why I suppressed it.

Reply:
Yeah you're right - it doesn't look like it is being used elsewhere at the moment - likely because it was put in when the footprint functions were being created.

However, with a nice generic function like this it would be useful to keep this broad as we will likely move to using it in other places (e.g. flux and boundary conditions would benefit from this) so would be good to keep as 2 separate functions.

Reply:
So for me the function `check_function_open_nc` is only used her https://github.com/openghg/openghg/blob/5eea4a4eb88a8581e44699cc3dd9ff31a9874951/openghg/standardise/footprints/_paris.py#L68 and here https://github.com/openghg/openghg/blob/5eea4a4eb88a8581e44699cc3dd9ff31a9874951/openghg/standardise/footprints/_acrg_org.py#L68 
(see https://github.com/search?q=repo%3Aopenghg%2Fopenghg%20check_function_open_nc&type=code)
But if you confirm @rt17603 that it need to be kept, I'll put it back.

The only purpose for me to use `check_function_open_nc` is to avoid `mypy` complaining at this stage. 
As we need to align the data in the case when calling `open_mfdataset`, I think it is better to align also the data while opening them when using `open_dataset` (for consistency).

Thus, another possibility would be to change `check_function_open_nc` to :
```
def check_function_open_nc(filepath: multiPathType, domain: str) -> Tuple[Callable, multiPathType]:
    """
    Check the filepath input to choose which xarray open function to use:
     - Path or single item List - use open_dataset
     - multiple item List - use open_mfdataset

    Args:
        filepath: Path or list of filepaths
        domain: domain of dataset
    Returns:
        Callable, Union[Path, List[Path]]: function and suitable filepath
            to use with the function.
    """
    import xarray as xr
    from functools import partial
    from openghg.util import align_lat_lon

    if isinstance(filepath, list):
        if len(filepath) > 1:
            xr_open_fn : Callable = partial(xr.open_mfdataset, preprocess=lambda x: align_lat_lon(x, domain))
        else:
            xr_open_fn = lambda x : align_lat_lon(xr.open_dataset(x), domain)
            filepath = filepath[0]
    else:
        xr_open_fn = lambda x : align_lat_lon(xr.open_dataset(x), domain)

    return xr_open_fn, filepath
    ```
    This seems to be ok from `mypy` point of view.

Reply:
Ah just saw your answer @rt17603 sorry. I'll keep them separated then

Reply:
The alternative in your comment looks good. Maybe we can have `open_and_align_data` as a function that calls `check_function_open_nc` and then returns `xr_open_fn(filepath)`? Then we could use `open_and_align_data` in `read_file` and so on?

Also, does `align_lat_lon` work if there are no lat/lon coords? Otherwise I think we need to distinguish what happens by data type.